### PR TITLE
709/from imports

### PR DESCRIPTION
**Pull request by aptenodytes-forsteri**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
- `from foo import bar` adds `//foo` as a dep instead of `//foo/bar`.
- `from foo.bar import baz` adds `//foo/bar` as a dep instead of `//foo/bar:baz` (when `//foo/bar:baz` is a separate py_library that contains foo/bar/baz.py as a source).
- `from google.cloud import aiplatform` does not add `"@gazelle_python_test_google_cloud_aiplatform//:pkg",` as a dep (fails to resolve `from` imports for pip packages).

Issue Number: #709 

## What is the new behavior?
When gazelle encounters `from foo.bar import baz`, try to resolve, in order from most specific to least specific:

```
foo.bar.baz
foo.bar
foo
```

This way, if `foo.bar.baz` is in the manifest file or in the index of existing rules (e.g. from a rule that contains the source file foo/bar/baz, it can be found). If `foo.bar.baz` can't be resolved, try `foo.bar` (maybe `baz` is just a function or variable in `foo/bar.py`).

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No - all existing tests continue to pass.

## Other information
May want to consider antagonistic cases - are there cases where imports are truly ambiguous?

### CODE REVIEW

File: gazelle/testdata/invalid_imported_module/test.yaml [hunk 1]
```diff
@@ -2,7 +2,8 @@
 expect:
   exit_code: 1
   stderr: |
-    gazelle: ERROR: failed to validate dependencies for target "//:invalid_imported_module": "grpc" at line 2 from "__init__.py" is an invalid dependency: possible solutions:
+    gazelle: Failed to find any valid modules from [grpc]. Errors: ["grpc" at line 2 from "__init__.py" is an invalid dependency: possible solutions:
```
Comment (aptenodytes-forsteri) [inline gazelle/testdata/invalid_imported_module/test.yaml:5]: I can clean up this error message

Comment (f0rmiga) [reply gazelle/testdata/invalid_imported_module/test.yaml:5]: Yeah, I wouldn't expect it to change.

Comment (aptenodytes-forsteri) [reply gazelle/testdata/invalid_imported_module/test.yaml]: Fixed.

**Review by f0rmiga — APPROVED** [review]
Thanks!!

**Pull request merged**